<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>G4_39</class>
 <widget class="QWidget" name="G4_39">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>735</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>G4_39</string>
  </property>
  <widget class="QTextBrowser" name="textBrowser">
   <property name="geometry">
    <rect>
     <x>24</x>
     <y>10</y>
     <width>761</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>20</pointsize>
    </font>
   </property>
   <property name="html">
    <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'幼圆'; font-size:20pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;题目：假设图书馆的图书包含书名、编号和作者属性，读者包含姓名、&lt;/span&gt;&lt;span style=&quot; font-family:'Calibri','sans-serif'; font-size:10.5pt;&quot;&gt;ID&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;和借书属性，每位读者最多可以借&lt;/span&gt;&lt;span style=&quot; font-family:'Calibri','sans-serif'; font-size:10.5pt;&quot;&gt;6&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;本书。编写程序&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt; font-weight:700; color:#ff0000;&quot;&gt;完成借书，还书，并列出&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;某读者的借书情况。&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>80</y>
     <width>191</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>书    本    录    入</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="book_writer">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>156</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>赫尔曼·黑塞</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="shang_jiao">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>314</y>
     <width>191</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>收      录</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>122</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>书名：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="book_name">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>116</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>荒原狼</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>160</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>作者：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="book_id">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>276</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>1</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>280</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>编号：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="stu_name">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>408</y>
     <width>131</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>张三</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="shang_jiao2">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>564</y>
     <width>181</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>借      阅</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>410</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>姓名：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>527</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>编号：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>490</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>书名：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="stu_book">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>488</y>
     <width>131</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>荒原狼</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="stu_id">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>448</y>
     <width>131</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>11122233</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>450</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>学号：</string>
   </property>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>348</y>
     <width>211</width>
     <height>20</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLineEdit" name="stu_book_id">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>525</y>
     <width>131</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>368</y>
     <width>191</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>读    者    借    阅</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QTabWidget" name="ji_lu">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>80</y>
     <width>541</width>
     <height>611</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="all_shuben">
    <attribute name="title">
     <string>书  本  信  息</string>
    </attribute>
    <widget class="QListView" name="all_book_xian">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>541</width>
       <height>591</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_2">
    <attribute name="title">
     <string>（ 读  者  / 书  本 ）借  阅  情  况</string>
    </attribute>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>幼圆</family>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>学号：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="stu_id2">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>18</y>
       <width>131</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>11122233</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="shang_jiao2_2">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>18</y>
       <width>51</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>幼圆</family>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>查询</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="book_id2">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>18</y>
       <width>131</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>荒原狼</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>22</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>幼圆</family>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>书名：</string>
     </property>
    </widget>
    <widget class="QPushButton" name="shang_jiao2_3">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>18</y>
       <width>51</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>幼圆</family>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>查询</string>
     </property>
    </widget>
    <widget class="QListView" name="reader_book">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>541</width>
       <height>531</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>198</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>国籍：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="book_nation">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>196</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>德国</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_14">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>700</y>
     <width>81</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>收录本数：</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="all_book_number">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>700</y>
     <width>101</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="ti_shi">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>700</y>
     <width>341</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="book_number">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>236</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>1</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_15">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>240</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>本数：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_16">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>620</y>
     <width>191</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>读    者    还    书</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="Line" name="line_2">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>600</y>
     <width>211</width>
     <height>20</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLineEdit" name="huan_book_idd">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>660</y>
     <width>131</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>1</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_17">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>662</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>编号：</string>
   </property>
  </widget>
  <widget class="QPushButton" name="shang_jiao2_4">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>700</y>
     <width>181</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>还      书</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>shang_jiao</sender>
   <signal>clicked()</signal>
   <receiver>G4_39</receiver>
   <slot>shou_lu()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shang_jiao2</sender>
   <signal>clicked()</signal>
   <receiver>G4_39</receiver>
   <slot>jie_yue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>186</x>
     <y>579</y>
    </hint>
    <hint type="destinationlabel">
     <x>243</x>
     <y>592</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shang_jiao2_2</sender>
   <signal>clicked()</signal>
   <receiver>G4_39</receiver>
   <slot>reader_cha()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>459</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>30</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shang_jiao2_3</sender>
   <signal>clicked()</signal>
   <receiver>G4_39</receiver>
   <slot>book_id_cha()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>724</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>788</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shang_jiao2_4</sender>
   <signal>clicked()</signal>
   <receiver>G4_39</receiver>
   <slot>huan_shu()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>713</y>
    </hint>
    <hint type="destinationlabel">
     <x>24</x>
     <y>707</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>shou_lu()</slot>
  <slot>jie_yue()</slot>
  <slot>reader_cha()</slot>
  <slot>book_id_cha()</slot>
  <slot>huan_shu()</slot>
 </slots>
</ui>
